File: POTD/8th_December_2024.cpp
class Solution {
  public:
    vector<vector<int>> mergeOverlap(vector<vector<int>>& arr) {
        // Code here
        vector<vector<int>>res;
        int n=arr.size();
        sort(arr.begin(),arr.end(),[](vector<int>&a,vector<int>&b){
            return a[0]<b[0];
        });
        int st=arr[0][0];
        int end=arr[0][1];
        for(int i=1;i<n;i++)
        {
            if(arr[i][0]<=end)
            end=max(end,arr[i][1]);
            else
            {
                res.push_back({st,end});
                st=arr[i][0];
                end=max(arr[i][1],end);
            }
        }
        res.push_back({st,end});
        return res;
    }
};

File: POTD/8th_December_2024.java
class Solution {
    public List<int[]> mergeOverlap(int[][] arr) {
        // Code here // Code here
        int n=arr.length;
        List<int[]>res=new ArrayList<>();
        Arrays.sort(arr,(a,b)->Integer.compare(a[0],b[0]));
        //Arrays.sort(arr,(a,b)->a[0]-b[0]);
        int st = arr[0][0];
        int end = arr[0][1];
        for(int i=1;i<n;i++){
            int[] current=arr[i];
            if(current[0]<=end){
                end=Math.max(end,current[1]);
            }else{
                res.add(new int[]{st,end});
                st=current[0];
                end=current[1];
            }
        }
        res.add(new int[]{st, end});
        return res;
    }
}
